http/codegen/service_data.go

```diff
@@ -1710,7 +1783,7 @@ func collectUserTypes(dt design.DataType, cb func(design.UserType), seen ...map[
 	}
 }
 
-func attributeTypeData(ut design.UserType, req, ptr, server bool, scope *codegen.NameScope, rd *ServiceData) *TypeData {
+func attributeTypeData(ut design.UserType, req, ptr, server, mustValidate bool, scope *codegen.NameScope, rd *ServiceData) *TypeData {
```

Comment:
Could we make sure the type has no required validation so we wouldn't need that extra argument? or do we need it for other reasons?

Reply:
Well, the validation code can generate validation for other things in the client and server types. But for an endpoint using a result type with multiple views, we don't use the validation code in the server and client packages, but use the validation logic generated in the views package. So this `bool` helps in not generating the validation code for them in the server and client packages.